src/goto-programs/loop_unroll.cpp

```diff
@@ -0,0 +1,213 @@
+#include <goto-programs/loop_unroll.h>
+
+bool unsound_loop_unroller::runOnLoop(loopst &loop, goto_programt &goto_program)
+{
+  int bound = get_loop_bounds(loop);
+  if(bound <= 0)
+    return false; // Can't unroll this
+
+  // Get loop exit goto number
+  goto_programt::targett loop_exit = loop.get_original_loop_exit();
+  // TODO: If we want a sound version: loop_exit++ so we have the skip instruction
+  if(loop_exit != goto_program.instructions.begin())
+  {
+    goto_programt::targett t_before = loop_exit;
+
+    if(t_before->is_goto() && is_true(t_before->guard))
+    {
+      // no 'fall-out'
+    }
+    else
+    {
+      // guard against 'fall-out'
+      goto_programt::targett t_goto = goto_program.insert(loop_exit);
+
+      t_goto->make_goto(loop_exit);
+      t_goto->location = loop_exit->location;
+      t_goto->function = loop_exit->function;
+      t_goto->guard = gen_true_expr();
+    }
+  }
+  /* TODO: For sound version
```

Comment:
Why do we want to keep these comments?

Reply:
I mostly maintain those to help me in the future when generalizing this into another class. But I will remove the code itself